lmfdb/templates/homepage.html

```diff
@@ -257,12 +257,13 @@ <h1 class="properties-header">Properties</h1>
         function cleanSubmit(id)
         {
             var myForm = document.getElementById(id);
-            var allInputs = myForm.getElementsByTagName('input')
-            var allSelects = myForm.getElementsByTagName('select')
-            var item, i;
+            var allInputs = myForm.getElementsByTagName('input');
+            var allSelects = myForm.getElementsByTagName('select');
+            var item, i, n = 0;
 
-            for(i = 0; item = allInputs[i]; i++) if(item.getAttribute('name') && !item.value) item.setAttribute('name', '');
-            for(i = 0; item = allSelects[i]; i++) if(item.getAttribute('name') && !item.value) item.setAttribute('name', '');
+            for(i = 0; item = allInputs[i]; i++) { if (item.getAttribute('name') ) { if (!item.value) { item.setAttribute('name', ''); } else { n++ }; } }
+            for(i = 0; item = allSelects[i]; i++) { if (item.getAttribute('name') ) { if (!item.value) { item.setAttribute('name', ''); } else { n++ }; } }
+            if (!n) { var all = document.createElement('input'); all.type='text'; all.name='all'; all.value='1'; myForm.appendChild(all); }
```

Comment:
@AndrewVSutherland , do you recall the purpose of this line?

Reply:
Not off the top of my head (it was a year ago), but I do remember it being necessary in some edge cases.

Reply:
Thanks!
@alexjbest figured it out.
It is the difference between http://lmfdb.xyz/EllipticCurve/Q/?all=1 vs  http://lmfdb.xyz/EllipticCurve/Q/?